pkgs/tools/networking/oama/default.nix

```diff
@@ -0,0 +1,97 @@
+{ mkDerivation
+, aeson
+, base
+, bytestring
+, containers
+, directory
+, fetchgit
+, hsyslog
+, http-conduit
+, lib
+, mtl
+, network
+, network-uri
+, optparse-applicative
+, pretty-simple
+, process
+, streaming-commons
+, string-qq
+, strings
+, text
+, time
+, twain
+, unix
+, utf8-string
+, warp
+, yaml
+}:
+let version = "0.14"; in
+mkDerivation {
+  pname = "oama";
+  inherit version;
+  src = fetchgit {
+    url = "https://github.com/pdobsan/oama.git";
+    sha256 = "1hdhkc6hh4nvx31vkaii7hd2rxlwqrsvr6i1i0a9r1xlda05ffq0";
+    rev = version;
+    fetchSubmodules = true;
+  };
+  isLibrary = true;
+  isExecutable = true;
+  libraryHaskellDepends = [
+    aeson
+    base
+    bytestring
+    containers
+    directory
+    hsyslog
+    http-conduit
+    mtl
+    network
+    network-uri
+    optparse-applicative
+    pretty-simple
+    process
+    streaming-commons
+    string-qq
+    strings
+    text
+    time
+    twain
+    unix
+    utf8-string
+    warp
+    yaml
+  ];
+  executableHaskellDepends = [
+    aeson
+    base
+    bytestring
+    containers
+    directory
+    hsyslog
+    http-conduit
+    mtl
+    network
+    network-uri
+    optparse-applicative
+    pretty-simple
+    process
+    streaming-commons
+    string-qq
+    strings
+    text
+    time
+    twain
+    unix
+    utf8-string
+    warp
+    yaml
+  ];
+
```

Comment:
isn't here needed to be:
```Nix
meta = with lib; {
<> 
};
```
?

Reply:
The `haskellPackages` version of `mkDerivation` doesn't like that, and instead all the attributes that would normally go in `meta` go at the top level.